modules/common/src/main/java/com/ibm/cloud/cloudant/internal/DelegatingAuthenticatorFactory.java

```diff
@@ -57,6 +56,6 @@ static Authenticator getAuthenticator(String serviceName,
         }
 
         // For all other auth types, delegate to the core factory
-        return ConfigBasedAuthenticatorFactory.getAuthenticator(serviceName);
+        return ConfigBasedAuthenticatorFactory.createAuthenticator(serviceProperties);
     }
 }
```

Comment:
I think it would be valuable to provide the dotted property mapper here too so we can avoid duplicating that functionality in other places; something like:
```suggestion
    static Authenticator getAuthenticator(Map<String, String> dottedProperties) {
        return createAuthenticator(
            dottedProperties
                .entrySet()
                .stream()
                .filter(entry -> entry.getKey().startsWith("cloudant."))
                .collect(Collectors.toMap(
                    // remove the cloudant. prefix and then uppercase and replace . with _
                    entry -> entry.getKey().substring(9).toUpperCase(Locale.ENGLISH).replaceAll("\\.", "_"),
                    entry -> entry.getValue())));
    }
}
```
and I suppose we ought to add a test or two for that.